marshalable-derive/src/lib.rs

```diff
@@ -66,21 +66,24 @@ fn derive_tpm_marshal_inner(input: DeriveInput) -> Result<TokenStream> {
             ));
         }
     };
-
     let expanded = quote! {
-        #pure_impl
-        // The generated impl.
-        impl Marshalable for #name  {
-            fn try_unmarshal(buffer: &mut UnmarshalBuf) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<Self> {
-                #unmarsh_text
-            }
+        #[doc(hidden)]
+        const _: () = {
+            use tpm2_rs_marshalable::__private::*;
```

Comment:
the #[doc(hidden)], anon module are not necessary. because you are trying to hide the fact that a trait is being implemented (at last that what code feel like it is trying). But that is not possible in rust.

As for the `use` I think it might not be necessary if we stick to the rule of fully qualified names

Reply:
I did that because serde did it and didn't question it, but you're right we don't need it (and likely they don't either - it came from a time before they used `_` to name these code blocks).

I'm not trying to hide the trait implementation, I'm trying to hide the const block, but on second look that seems unneeded.

Reply:
Apparently blames to this: https://github.com/serde-rs/serde/commit/a135199ab104a8068f6dd672f813760d17128212

Anyways, I'll remove this.